sklearn/metrics/ranking.py

```diff
@@ -100,7 +100,11 @@ def auc(x, y, reorder=False):
                                  "the x array is not increasing: %s" % x)
 
     area = direction * np.trapz(y, x)
-
+    if isinstance(area, np.memmap):
+        # Reductions such as .sum used internally in np.trapz do not return a
+        # scalar float by default for numpy.memmap instances contrary to
+        # traditional arrays.
+        area = float(area)
```

Comment:
Not sure that's the best way to do it but I found:

```python
area = area.dtype.type(area)
```

or

```python
area = area[()]
```
to return a numpy float rather than a Python float.